### PR TITLE
Binary deletion and doc preparation

### PR DESCRIPTION
**Pull request by tlaurion**
# Description
Read all the README.md
* Binaries got deleted.
* Documentation got segmented
* Test report under dhcp-forwarder

# Impacts
replaces udp-reflector for Windows, reduces future IP Helper configuration, mitigate 5/8 of DHCP traffic from tests ran on a 50000 packets capture.

# Delete branch after merge
YES 

# UPGRADE file entries
Replaces udp-reflector for Windows.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,72 +42,156 @@ Does:
...
+You now have an installer under "c:\go\src\packetfence-dhcp-forwarder\dhcp-forwarder-installer\DHCP Forwarder installer.exe"
+
+
+Troubleshoot
+============
+Eventlog
+--------
+The Event logs should help a lot in finding the cause the service not starting. Have you changed your networking card since Installation? Disconnected a cable disconnected? Had the server sleep and resumed from suspend?
+
+Alternatively, you can stop the service from wWindows Service Manager and debug from the command line. Launch an adminstrative command line and place yourself under "C:\Program Files (x86)\DHCP Forwarder"
```
Comment (extrafu) [inline README.md]: wWindows -> Windows

File: README.md [hunk 2]
```diff
@@ -1,21 +1,39 @@
...
-3. Run dhcp-forwarder-config-generator.exe which generates a configuration file in previous directory.
-4. Install dhcp-forwarder.exe as a service with the help of nssm
-5. Start dhcp-forwarder.exe with the help of nssm.
+This tool captures and forwards a subset of DHCP traffic (specifically DHCPREQUEST and DHCPACK) from a Windows DHCP server to a destination IP and port.
+
+Alternatively, IP Helpers can be configured on each switch of an infrastructure to forward broadcast only packets. Those contain all types of DHCP packets but less to none DHCPACK which confirms lease ownership.
+
+DHCP traffic is useful to PacketFence to link MAC adresses and IP addresses, while also helping Fingerbank's fingerprinting process. Again, the only useful packets to PacketFence are DHCPREQUEST and DHCPACK.
+
+In short, if DHCP-Forwarder can be deployed centrally, it should be done. In that case, only useful packets are captured and forwarded from the source, which reduces configuration, transport, processing and storage costs of useless packets, while supressing the need to process them to actually neglect them at the destination host.
```
Comment (extrafu) [inline README.md]: supressing (typo here) -> removing

File: README.md [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
+
+Binaries
+========
+dhcp-forwarder.exe
+------------------
+This tool captures and forwards DHCP traffic (DHCPREQUEST and DHCPACK, specifically) from a Windows DHCP server to PacketFence. 
 
-The installer is downloadable from: 
-dhcp-forwarder-installer\DHCP Forwarder Installer.exe
+DHCPREQUEST and DHCPACK packets are the ones being the most important for PacketFence to link MAC to IP addresses and switch location and help Fingerbank to fingerprint the Operating System running on the device.
```
Comment (extrafu) [inline README.md]: Operating System -> operating system

File: README.md [hunk 4]
```diff
@@ -1,21 +1,39 @@
...
+dhcp-forwarder.exe
+------------------
+This tool captures and forwards DHCP traffic (DHCPREQUEST and DHCPACK, specifically) from a Windows DHCP server to PacketFence. 
 
-The installer is downloadable from: 
-dhcp-forwarder-installer\DHCP Forwarder Installer.exe
+DHCPREQUEST and DHCPACK packets are the ones being the most important for PacketFence to link MAC to IP addresses and switch location and help Fingerbank to fingerprint the Operating System running on the device. 
 
-dhcp-forwarder-installer\* contains all binaries packaged in the installer.
+This fingerprinting and localisation process helps a lot in determining violation triggers condition.
```
Comment (extrafu) [inline README.md:23]: triggers condition -> trigger conditions.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,39 @@
...
+ 
+With the help of DHCP-Forwarder, those DHCP packets can be obtained directly and easily from the source. 
 
+Alternatively, IP helpers can be configured on each switch to forward DHCP traffic to PacketFence, but only broadcast packets can be captured by them, which is less precise. Deploying DHCP-Forwarder is simple and centralized.
+
+DHCP-forwarder is based on gopacket and depends upon WinPCAP to select the requested packets through a BPF, which is really fast. Captured traffic is then forwarded to a configured host and port. 
+
+A default BPF is produced by the configuration generator. That filter can be manually modified in the configuration file by the user.
+
+DHCP-Forwarder requires a DHCP-Forwarder.toml file to be present in it's working directory (installation directory). DHCP-Forwarder.toml is generated from dhcp-forwarder-config-generator.exe at installation time, but can be run from the installation directory anytime.
```
Comment (extrafu) [inline README.md:33]: it's -> its

Comment (extrafu) [reply README.md:33]: run -> executed

File: README.md [hunk 6]
```diff
@@ -24,72 +42,161 @@ Does:
 4. Store default filter value, which selects DHCPACK and DHCPREQUESTS in a DHCP mask.
 5. Store those values in DHCP-Forwarder.toml in the working directory.
 
+Note: Do not select a Wireless device, it will not work.
```
Comment (extrafu) [inline README.md]: Wireless -> wireless

File: README.md [hunk 7]
```diff
@@ -24,72 +42,161 @@ Does:
...
+1. After a configuration change
+2. When the server goes to sleep and resumes
 
 
-Native Compilation under x64:
-===========
-You will need:
+The installer
+-------------
+The installer will:
```
Comment (extrafu) [inline README.md:55]: start each bullet with a lower-case letter, only the last line should have a trailing dot.

File: README.md [hunk 8]
```diff
@@ -24,72 +42,161 @@ Does:
...
-* Files are installed under "C:\Program Files (x86)\DHCP Forwarder".
+Files are installed under "C:\Program Files (x86)\DHCP Forwarder".
 
 
-Troubleshoot:
-===========
-Launch an adminstrative command line and place yourself under "C:\Program Files (x86)\DHCP Forwarder"
+Compilation
+-----------
+Once you have the sources and the tools for Native compilations under c:\go\src\
```
Comment (extrafu) [inline README.md]: Native compilations -> native compilation

File: README.md [hunk 9]
```diff
@@ -24,72 +42,161 @@ Does:
...
+
+
+You now have an installer under "c:\go\src\packetfence-dhcp-forwarder\dhcp-forwarder-installer\DHCP Forwarder installer.exe"
+
+
+Troubleshoot
+============
+Eventlogs
+--------
+The Event logs should help a lot in finding the cause the service not starting. Have you changed your networking card since Installation? Disconnected a cable disconnected? Had the server sleep and resumed from suspend?
```
Comment (extrafu) [inline README.md]: is not starting

Comment (extrafu) [reply README.md]: Installation -> installation

**Comment by julsemaan** [comment]
You'll need to rebase this off master

Looks like the binary is still part of this branch and was changed in the branch.

Just rebase and delete it during the rebase

**Comment by tlaurion** [comment]
Following command has been ran:
`git filter-branch --force --index-filter 'git rm --cached --ignore-unmatch "dhcp-forwarder-installer/DHCP Forwarder Installer.exe" "dhcp-forwarder-installer/DHCP Forwarder.exe" "dhcp-forwarder-installer/WinPcap_4_1_3.exe" "dhcp-forwarder-installer/dhcp-forwarder-config-generator.exe" "dhcp-forwarder-installer/dhcp-forwarder.exe"  "dhcp-forwarder-installer/nssm.exe" "dhcp-forwarder-config-generator/dhcp-forwarder-config-generator.exe" "dhcp-forwarder/dhcp-forwarder.exe"' --prune-empty --tag-name-filter cat -- --all`

Everyone will have to rebase.

**Pull request merged**